apps/site/assets/ts/stop/components/DeparturesFilters.tsx

```diff
@@ -1,79 +1,74 @@
-import { each, startCase, uniqueId } from "lodash";
-import Pill from "./Pill";
+import { sortBy, startCase } from "lodash";
+import React, { ReactElement } from "react";
 import { Mode } from "../../__v3api";
+import { iconForMode } from "../../models/route";
 
-export const ALL = "all";
-export const BUS: Mode = "bus";
-export const SUBWAY: Mode = "subway";
-export const FERRY: Mode = "ferry";
-export const COMMUTER_RAIL: Mode = "commuter_rail";
+export type ModeChoice = "all" | Mode;
+interface ModeDisplay {
+  displayText: string;
+  displayIcon?: JSX.Element;
+  mode: ModeChoice;
+}
 
-/* eslint-disable @typescript-eslint/no-explicit-any */
-const filterDeparturesByMode = (
-  departuresArray: any[],
-  mode: Mode | typeof ALL
-): any[] => {
-  if (mode === ALL) {
-    return departuresArray;
+const buttonNameForMode = (mode: Mode): string => {
+  if (mode === "commuter_rail") {
+    return "Rail";
   }
-  return departuresArray.filter(d => d.mode === mode);
+  return startCase(mode);
+};
+
+// Commuter Rail, then Subway, then Bus
+const buttonSortFn = (mode: Mode): number => {
+  if (mode === "commuter_rail") {
+    return 1;
+  }
+  if (mode === "subway") {
+    return 2;
+  }
+  if (mode === "bus") {
+    return 3;
+  }
+  return 4;
 };
 
 const DeparturesFilters = ({
-  departures,
-  onModeChange
+  modesList,
+  selectedMode,
+  setSelectedMode
 }: {
-  departures: any[];
-  onModeChange: (val: any[]) => void;
-}): ReactElement<HTMLElement> => {
-  /* eslint-enable @typescript-eslint/no-explicit-any */
-
-  const [selectedMode, setSelectedMode] = useState<typeof ALL | Mode>(ALL);
-  const [filterOptions, setFilterOptions] = useState<
-    { displayText: string; mode: typeof ALL | Mode }[]
-  >([]);
-
+  modesList: ModeChoice[];
+  selectedMode: ModeChoice;
+  setSelectedMode: (val: ModeChoice) => void;
+}): ReactElement<HTMLElement> | null => {
+  if (!modesList.length) return null;
   // Create filter list
-  useEffect(() => {
-    if (departures.length === 0) {
-      setFilterOptions([]);
-    } else {
-      const filterOptionsArray: {
-        displayText: string;
-        mode: typeof ALL | Mode;
-      }[] = [
-        {
-          displayText: startCase(ALL),
-          mode: ALL
-        }
-      ];
-      each([BUS, SUBWAY, COMMUTER_RAIL, FERRY], arrayMode => {
-        if (filterDeparturesByMode(departures, arrayMode).length > 0) {
-          filterOptionsArray.push({
-            displayText: startCase(arrayMode),
-            mode: arrayMode
-          });
-        }
-      });
-      setFilterOptions(filterOptionsArray);
+  const displayedModes: ModeDisplay[] = [
+    {
+      displayText: "All",
+      mode: "all" as ModeChoice
     }
-  }, [departures]);
-
-  useEffect(() => {
-    onModeChange(filterDeparturesByMode(departures, selectedMode));
-  }, [departures, selectedMode, onModeChange]);
-
+  ].concat(
+    sortBy(modesList, [buttonSortFn]).map(mode => ({
```

Comment:
Instead of starting with the `modesList` and sorting, what about starting with the full list of modes, and filtering based on their existence in `modesList`? That way the preferred order is laid out in code, and we wouldn't need to define sort keys.